src/main/java/com/google/sps/servlets/ForumServlet.java

```diff
@@ -0,0 +1,119 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.gson.Gson;
+import com.google.sps.data.ForumElement;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles forum data (questions, comments, and replies) */
+@WebServlet("/forum")
+public class ForumServlet extends HttpServlet {
+
+  /** Get request to the Datastore and responds with the forum elements with the parentId input */
```

Comment:
You're using "id", not "parentId".

Also, as Mike (and I) commented in Praneet's PR, these parameter names should be constants: https://github.com/googleinterns/granite2020/pull/12#discussion_r467073578

Reply:
I made the parameter names constants in the ForumElement.java class because with the new static factory they are used in factory method in ForumElement.java too.